src/App.jsx

```diff
@@ -14,33 +16,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_GOODS_ALPHABET = 'Alphabet';
+const SORT_GOODS_LENGTH = 'Length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortFild, reversed) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  preparedGoods.sort((good1, good2) => {
+    switch (sortFild) {
+      case SORT_GOODS_ALPHABET:
+        return good1.localeCompare(good2);
+      case SORT_GOODS_LENGTH:
+        return good1.length - good2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortFild, setSortFild] = useState('');
```

Comment:
I think `fild` is a typo

I recommend using the following VS Code extension that will help you identify this type of mistake:
https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker